mlrun/package/constants.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+class ArtifactTypes:
```

Comment:
Enums require doing `x.value` on them to get the string value, and because users may define their own types to do so (for example: `pack_imageset` for artifact type `"imageset"`) I decided to remove the `Enum` and leave it as a collection of commonly used options. Same for log hints.

Regarding the plural use, I'll change that :+1:

Reply:
Well, you don't have to use the string representation for comparison, for example, you can do -
```assert my_value == ArtifactType.SOME_TYPE```, for example (the `Enum` class actually creates a singleton per included value). You only need string comparison if you serialize the value.
Also, what do you mean by "their own types"? I thought these types are only for artifacts, so users are not expected to just add a new artifact type out of the blue.

Reply:
* The value is serialized in the function yaml and the runtime.
* We have Artifact, DirArtifact, DatasetArtifact and ModelArtifact (the plots are basically artifacts with body....) so the user can decide to add a `pack_image_dataset` to special handle how to log images datasets, so he could use a log hint like: `"my_imageset: image_dataset"` for example. It's not really an artifact type by technical MLRun definition but it can be seem like so by a user of MLRun.